test/easyhtml_test.exs

```diff
@@ -57,4 +57,28 @@ defmodule EasyHTMLTest do
     html = ~HTML[<p id="p1">Hello, <em>world</em>!</p>]
     assert inspect(html) == ~s|~HTML[<p id="p1">Hello, <em>world</em>!</p>]|
   end
+
+  test "enumarble" do
+    html = ~HTML[
+      <table>
+        <tr>
+          <td>A</td>
+          <td>1</td>
+        </tr>
+        <tr>
+          <td>B</td>
+          <td>2</td>
+        </tr>
+      </table>
+    ]
+
+    result =
+      for tr <- html["tr"], td <- tr["td"] do
+        assert is_struct(tr, EasyHTML)
+        assert is_struct(td, EasyHTML)
+        to_string(td)
+      end
+
+    assert result == ["A", "1", "B", "2"]
```

Comment:
could you write the following tests instead?

```suggestion
    assert Enum.list(html) == [html]

    assert Enum.to_list(html["tr"]) == [~HTML[...], ~HTML[...]]

    assert Enum.to_list(html["bad"]) == []
```

Reply:
Sure, I can do that. This
```elixir
 assert Enum.to_list(html["bad"]) == []
 ```
will not work though since EasyHTML returns `nil`
 ```elixir
 html["bad"] # => nil
 ```
 
 
 ```
 ** (Protocol.UndefinedError) protocol Enumerable not implemented for nil of type Atom. This protocol is implemented for the following type(s): Date.Range, EasyHTML, File.Stream, Floki.HTMLTree, Function, GenEvent.Stream, HashDict, HashSet, IO.Stream, List, Map, MapSet, Range, Stream
     code: assert Enum.to_list(html["bad"]) == []
```

Reply:
oh, right, nevermind that one then.

Reply:
I changed the tests.